app/bundles/LeadBundle/Entity/TagRepository.php

```diff
@@ -91,18 +89,38 @@ public function checkLeadByTags(Lead $lead, $tags)
 
         $q = $this->_em->getConnection()->createQueryBuilder();
         $q->select('l.id')
-            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l');
-        $q->join('l', MAUTIC_TABLE_PREFIX.'lead_tags_xref', 'x', 'l.id = x.lead_id')
-                ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags', 't', 'x.tag_id = t.id')
-                ->where(
-                    $q->expr()->andX(
-                        $q->expr()->in('t.tag', ':tags'),
-                        $q->expr()->eq('l.id', ':leadId')
-                    )
+            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
+            ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags_xref', 'x', 'l.id = x.lead_id')
+            ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags', 't', 'x.tag_id = t.id')
+            ->where(
+                $q->expr()->andX(
+                    $q->expr()->in('t.tag', ':tags'),
+                    $q->expr()->eq('l.id', ':leadId')
                 )
-                ->setParameter('tags', $tags, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
-                ->setParameter('leadId', $lead->getId());
+            )
+            ->setParameter('tags', $tags, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
+            ->setParameter('leadId', $lead->getId());
 
         return (bool) $q->execute()->fetchColumn();
     }
+
+    /**
+     * Save an entity through the repository.
+     *
+     * @param object $entity
```

Comment:
Can you add correct typehint?

Reply:
Overridden methods can change param types?

Reply:
Why not? It is only a doc block.

Reply:
In doc bock? Yep, I can do that.